pynamodb/tests/test_attributes.py

```diff
@@ -707,17 +722,69 @@ def __eq__(self, other):
         person1 = Person()
         person1.name = 'john'
         person1.age = 40
+        person1.extra_arbitrary_data = {'married': True}
 
         person2 = Person()
         person2.name = 'Dana'
         person2.age = 41
+        person2.extra_arbitrary_data = {'net_income': 200000,
+                                        'dog': 'Fido'}
+
         inp = [person1, person2]
 
         list_attribute = ListAttribute(default=[], of=Person)
         serialized = list_attribute.serialize(inp)
         deserialized = list_attribute.deserialize(serialized)
         self.assertEqual(sorted(deserialized), sorted(inp))
 
+    def test_list_of_map_with_of_and_custom_attribute(self):
+
+        # Create a couple of mock functions to use
+        # to test that the CustomAttribute serialize/deserialize are called
+        serialize_mock = Mock()
+        deserialize_mock = Mock()
+
+        class CustomAttribute(Attribute):
+            attr_type = STRING
+
+            def serialize(self, value):
+                serialize_mock(value)
+                return value.upper()
+
+            def deserialize(self, value):
+                deserialize_mock(value)
+                return value.lower()
+
+        class CustomMapAttribute(MapAttribute):
+            custom = CustomAttribute()
+
+            def __eq__(self, other):
+                return self.custom == other.custom
+
+        attribute1 = CustomMapAttribute()
+        attribute1.custom = 'test-value1'
+
+        attribute2 = CustomMapAttribute()
+        attribute2.custom = 'test-value2'
+
+        inp = [attribute1, attribute2]
+
+        list_attribute = ListAttribute(default=[], of=CustomMapAttribute)
+        serialized = list_attribute.serialize(inp)
+        deserialized = list_attribute.deserialize(serialized)
+        assert deserialized == inp
+
+        # Confirm that the the serialize/deserialize are called
+        # with the expected values
+        serialize_mock.assert_has_calls([
```

Comment:
Unfortunately the version of mock we're using is pretty old and doesn't have methods like these. Can you rebase off https://github.com/pynamodb/PynamoDB/commit/4f56221d7860e3f27f129963a40f96034a3f14f3 so we can be sure these are passing? Thanks!